app/models/effects/external.rb

```diff
@@ -0,0 +1,48 @@
+module Effects
+  class External < Effect
+    class ExternalEffectFailed < StandardError; end
+
+    def perform(workflow_id, subject_id)
+      reductions = SubjectReduction.where(
+        workflow_id: workflow_id, 
+        subject_id: subject_id, 
+      )
+      reductions = reductions.where(reducer_key: config[:reducer_key]) if config[:reducer_key]
+
+      begin
+        if valid?
+          response = RestClient.post(url, reductions.to_json, {content_type: :json, accept: :json})
+        else
+          raise StandardError.new "External effect has missing or invalid URL"
+        end
+      rescue RestClient::InternalServerError
+        raise ExternalEffectFailed
+      end
+    end
+
+    def valid?
+      url.present? && valid_url?
+    end
+
+    def self.config_fields
+      [:url].freeze
```

Comment:
Shouldn't this contain the reducer_key as well?

Reply:
✔️